### PR TITLE
Clone of apple.com website - Solo Project

### PR DESCRIPTION
**Pull request by ybadmus**
The project is a clone of the official apple.com website. The real project can be found here: _https://web.archive.org/web/20140301004610/http://www.apple.com/_. This project is a solo project undertaken as part of the HTML & CSS section of Microverse program.

### CODE REVIEW

**Comment by majovanilla** [comment]
Hey @ybadmus , 👋

## Good job on this project! 👍

There are still some changes 👇 you need to implement to approve this request 🔍

### Required Changes: ✍️

- Always remember to check your HTML using the [validator](https://validator.w3.org/). You have some warnings. Please fix them.

- Always remember to activate your repo in Stickler. I cannot see your linters working!!

- Improve your Readme file. It should be descriptive of the project. Here's a great [template](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/) I strongly recommend you to use.

- Improve your Readme file. It should be descriptive of the project. You are using the correct template but now you need to fill it with information about your project. You can add details about what you did in this project, why you did it, how you did it... Also, you have a lot of subtitles empty. If you are not going to use them, then delete them.

-Please write meaningful pull request messages. This is a very important step of your review process, please fix it. Check this [example](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/)

**Navbar**

- The width and height of your navbar is not the same as the original, please fix this.

- You should work on your gradient to make it look similar to the one on the original page. You are using the gradient to the right, but that is not how you can achieve the effect on the original page.

- You are missing the search bar on your navbar. Please include it and make it look like the original one.

**Main**

- You are using a `<header>` tag to enclose the main titles. This is not a good practice since you already have an element before that. Check this [article](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header) to see more about this tag and how to use it.

**Footer**

- Please modify the style of your links. The ones on the left should have a different color. 

- The elements on the right should be links and be separated by a vertical gray line.

### Optional Changes:

If you want to make your project even better 🔥💥, I have a few suggestions you can implement.  

These are completely [Optional]: 🦔

- Use semantic HTML in the structure of your HTML file. Favor the use of `<nav>`,  `<main>`,  `<footer>`, to enclose the elements on your file. 

- Your project would look sooo much better if you include the hover effects on the links of the page, like the original version.

## Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

**Comment by ybadmus** [comment]
Dear Maria,

You made these suggestion concerning the navbar

*Navbar*

   -

   The width and height of your navbar is not the same as the original,
   please fix this.
   -

   You should work on your gradient to make it look similar to the one on
   the original page. You are using the gradient to the right, but that is not
   how you can achieve the effect on the original page


I have tried a couple of things but not working, can you please explain
further how to achieve these. It will be very helpful.

Moreover, I am still having issues with stickler even after activating it.
Please crosscheck and advise.

repo: https://github.com/ybadmus/apple-clone/pull/1

Best Regards,
Yusuf

**Review by nikasulo — CHANGES_REQUESTED** [review]
# Under Review By Tolulope Olusakin(TSE)
Please check on this in a few minutes for some feedback! :+1:

**Review by nikasulo — CHANGES_REQUESTED** [review]
Great work @ybadmus! :clap: 
You have done a really nice job putting this all together :+1: Please consider my comments below, make the necessary changes and request another review. I wish you all the best and see you at the top :100: 

#### Navbar
Please consider comparing your navigation bar with the original, you may find that it's a little different in size, positioning and some pieces may be missing. Please look closely and make the necessary adjustments.

#### Horizontal Scroll
You seem to have a little bug with one of the elements of the page, shooting out beyond the browser's maximum width. This may be as a result of a margin or padding you have set up. One of the easiest ways I have found to correct this issue is to set the default `box-sizing` property to `border-box` as follows `*{box-sizing: border-box}`. Please read some more about it [here](https://developer.mozilla.org/en-US/docs/Web/CSS/box-sizing).

#### Use of background images
Please consider that as per the requirements for this project, the images in the footer are supposed to be `<div>` tags with their `background-image` properties set to the image URL and not image tags inside divs
screenshot

##### Think I missed something? Need some help with this or any of your projects? Please feel free to reach out to me on [Twitter](https://twitter.com/oluwadamilareo_) or [Instagram](https://instagram.com/oluwadamilare_olusakin), even if you just want to chat and I will respond to you as soon as I can.

**Comment by addod19** [comment]
## Pull Request under review by Daniel Larbi Addo (TSE)

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes ♻️ 🚥 

Hi @ybadmus, you did a great job fixing the changes recommended by the previous TSE's 👍 However, we left out some of the required changes. Find comments below:
* Kindly edit the PR, we cannot approve the project with it.
PR
Click on the ```...``` icon next to the 😃  icon and make the necessary changes and pointed out by the first TSE 😄 
* Kindly fix the horizontal scroll pointed out the by second TSE
* Kindly apply the DRY rule to the CSS file so we avoid code repetition. Check [here](https://speckyboy.com/good-bad-css-practices/)
* Kindly update the live link with the project live link

Kindly make these changes and submit another code review 👍 
_Happy coding_ 👨‍💻 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Comment by ybadmus** [comment]
Thanks for you review, I just updated changes.

Kindly check and approve.

Best Regards!
Yusif

**Review by christianotieno — COMMENTED** [review]
## Pull Request under review by Christian Otieno (TSE)

File: README.md
Comment (christianotieno) [inline README.md]: # [REQUIRED] - FIX - Screenshot
I am unable to see the screenshot from my end.
This is what I mean: 👇 
screenshot

 so kindly fix it with a dot URL as shown here
```suggestion
![screenshot](./app_screenshot.png)
```

File: README.md
Comment (christianotieno) [inline README.md]: # [REQUIRED] - UPDATE - The README file
Your README file is not as well descriptive as it out to professionally be.
- There are no links to your live preview and the rest of the links are not working as well.
- Kindly update your README and remove all the unnecessary infomation in the file. 
- You may want to follow these [guidelines](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/) as an inspiration on how to do well.

File: index.html
Comment (christianotieno) [inline index.html]: # [REQUIRED] - REMOVE - The Styling in your  markup
Move all these stylings to the CSS file

File: index.html
Comment (christianotieno) [inline index.html]: # [REQUIRED] - FIX - errors
Your markup is still not passing the validation checks. 
html_validator_not_passed
Kindly do help fix that.

**Review by christianotieno — CHANGES_REQUESTED** [review]
Christian Otieno  9:55 PM
# **CHANGES REQUIRED** :warning:
Hey @ybadmus  :wave:
-
You have done well with the changes! :tada: You are almost there.
We, however, do have some few changes to carry out before we proceed (see review comment).
##

Please makes changes and submit another review request.
- If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email. :eyes:
##
- When you have made the specified changes, please submit another code review. Thank you!
Happy coding 💯 
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)
---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Comment by ybadmus** [comment]
All changes have now been resolved, kindly review and approve. GitHub pages source is set to the master branch and I have not merged into it yet, this is the reason why the live link only show the title of the project. This will be resolved automatically after this project is approved and I merge into the master branch.

Best Regards!
Yusif

**Comment by mikenath223** [comment]
### Pull request currently under review by Michgolden Ukeje (TSE).
Kindly exercise patience 🐱

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi @ybadmus 👋

You've done quite well on this project👏

Your project looks almost good to go.
But however, a few more changes need to be met.

You are almost there, just hang in there you've got this.💪

Kindly make the following changes:👇

- [ ] Required - Check your HTML for validation errors with this [tool](https://validator.w3.org/nu/) and fix it.

- [ ] Required - Also remove all comments in your CSS code. If you want to refer to an earlier version of your code, consider using git history.

- [ ] Required - Consider the way the navbar looks on the original webpage and compare it to yours.**Then make the changes listed below**
**THIS IS YOURS👇**
ice_screenshot_20200402-132803

**THIS IS THE WAY IT SHOULD LOOK**👇
ice_screenshot_20200402-135135


* Make the following changes to your nav-bar:

- Add a search icon to the search input field as seen in the original.
- Change the background color of the `search input` field to look like that of the original.
-  Add a radius to the nav-bar's border as seen in the original.

* NOTE - Whenever your project requires node_modules kindly consider adding it to .gitignore **before you make your first commit.**

Kindly fix these issues and request another review👍
 
You are almost there, you've got this💪😊

Feel free to contact me on slack(michgolden) if you need further clarification.

HAVE A SMASHING CODING DAY! 👨‍💻💜

**Comment by Dandush03** [comment]
# :red_circle: Status: Required Changes :red_circle:
### Hey @ybadmus :wave: 
### Good attempts on this one :muscle: :+1: 

## Your project looks great! I see that you try to address all the others TSE corrections but your projects is breaking after your new implementation  :point_down: 

- [ ]  :one: Your menu should not occupy the whole image width 

#### Required:
Capture
#### Yours:
Capture2

- [ ]  :two: Your bottom images should be aligned in the center
#### Required:
Capture3
#### Yours:
Capture1


## Kindly address all these changes and submit a new code review

### Happy coding :computer: 

#### Daniel Laloush

**Comment by ybadmus** [comment]
This is not the case with other browsers. Did you check this with chrome?


I will implement the changes anyway.

Best Regards,
Yusif

**Review by kevcap — APPROVED** [review]
# [STATUS] PROJECT APPROVED 🍾

### I really like your project! Congrats, you did an excellent job implementing the required changes and you have my approval to go ahead and merge this pull request!

_**Clean and modular code, following the best practices and the design looks great, keep it up!**_ 🥇

_Note_: That changes marked as [OPTIONAL] are not required to approve your project, but I strongly recommend that you guys go for that extra mile, to get more knowledge and have a better project to add at your portfolio.

## [OPTIONAL CHANGES]
**⚠These are mandatory changes, but since It wasn't pointed out earlier, I'll mark them as [OPTIONAL]:**
- [ ] Files not related to the project should not be committed and added to .gitignore (e.g. Node Modules file ) ([screenshot ](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png)from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

## As always, I wish you all the very best and see you both at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

**Comment by ybadmus** [comment]
Well noted!

Thanks

**Pull request merged**
